assignment/scripts/array-practice.js

```diff
@@ -4,77 +4,87 @@ console.log('****** Array Practice *******');
 
 // Example: How many animals are in the array? 
-console.log('Number of animals:', animalArray.length);
+// console.log('Number of animals:', animalArray.length);
 
 // 2.a. TODO: Log to the console the number of foods in your array
-
+console.log('Number of favorite Italian desserts:', favoriteFoods.length);
 
 // 3. Accessing array items
 console.log('--- 3. Accessing items in an array ---');
 
 // Example: Log the first animal from the array using it's array index
-console.log('First animal is', animalArray[0])
-
-// 3.a. TODO: Log the second animal in the array 
-
+// console.log('First animal is', animalArray[0])
 
-// 3.b. TODO: Log the last animal in the array using it's array index 
+// 3.a. TODO: Log the second food in the array 
+console.log('Second dessert is:', favoriteFoods [1]);
 
+// 3.b. TODO: Log the last food in the array using it's array index 
+console.log('Last dessert is:', 'rainbow cookie');
 
-// 3.c. (STRETCH) TODO: Log the last animal by using the array length, 
+// 3.c. (STRETCH) TODO: Log the last food by using the array length, 
 //      instead of the exact index number of the last item
-
-
+let lastFood = favoriteFoods[favoriteFoods.length - 1];
+console.log(lastFood);
 
 // 4. Adding & Removing Array Items
 console.log('--- 4. Adding and removing array items ---');
 
 // Example: Add an animal to the end using Array.push
-animalArray.push('penguin');
-console.log('Added an animal to end,', animalArray);
+// animalArray.push('penguin');
+// console.log('Added an animal to end,', animalArray);
 
 // 4.a. TODO: Add a new food at the end of your array & log the array
-
-
+favoriteFoods.push('biscotti');
+console.log('Added a new favorite Italian dessert:', favoriteFoods);
 
 // Example: Remove the last animal by using Array.pop
-let removedAnimal = animalArray.pop();
-console.log('Removed the last animal', removedAnimal);
-console.log('The animals are now', animalArray);
+// let removedAnimal = animalArray.pop();
+// console.log('Removed the last animal', removedAnimal);
+// console.log('The animals are now', animalArray);
 
 // 4.b. TODO: Remove the food at the end of your array & 
 //      log both the food removed and the updated array
-
+let removedFood = favoriteFoods.pop();
+console.log('Removed the last food,', removedFood);
+console.log('The favorite Italian desserts are now:', favoriteFoods);
 
 // Example: Add an animal to the beginning using Array.unshift
-animalArray.unshift('walrus');
-console.log(`Added an animal to beginning: ${animalArray}`);
+// animalArray.unshift('walrus');
+// console.log(`Added an animal to beginning: ${animalArray}`);
 
 // 4.c. TODO: Add a food at the beginning of the array & log the array
+favoriteFoods.unshift('tiramisu');
+console.log('Added a new dessert to the beginning:', favoriteFoods);
 
+// Hey Zac, how is the above different from just logging this:
+// console.log(`Added a new dessert to the beginning: ${favoriteFoods}`);
+// like in the example?
```

Comment:
The difference is subtle, though you could spot it in your console.

Using a template string actually creates a single string value. So this:

```js
`Added a new dessert to the beginning: ${favoriteFoods}`
```

Becomes this:

```js
"Added a new dessert to the beginning: cannoli,italian ice,sfogliatella,rainbow cookie"
```

The array has to get converted into a string, which JavaScript does by putting a comma between each element.

If you instead pass the array into `console.log`, nothing is converted. The `console.log` function can take a variety of inputs and has different ways of showing them in a console. For an array, you should see it as a collapsible/expandable array object.